src/main/scala/com/fulcrumgenomics/phasing/HapCutToVcf.scala

```diff
@@ -0,0 +1,413 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package com.fulcrumgenomics.phasing
+
+import java.util
+import java.util.NoSuchElementException
+
+import com.fulcrumgenomics.cmdline.{ClpGroups, FgBioTool}
+import dagr.commons.CommonsDef._
+import dagr.commons.io.Io
+import dagr.commons.util.LazyLogging
+import dagr.sopt._
+import htsjdk.samtools.util.CloserUtil
+import htsjdk.variant.variantcontext._
+import htsjdk.variant.variantcontext.writer.{Options, VariantContextWriter, VariantContextWriterBuilder}
+import htsjdk.variant.vcf._
+
+import scala.collection.JavaConversions._
+import scala.collection.JavaConverters._
+
+@clp(
+  description =
+    """
+      |Converts the output of HapCut to a VCF.
+      |
+      |The output of HAPCUT does not include all input variants, but simply those variants that are in phased blocks.
+      |This tool takes the original VCF and the output of HapCut, and produces two VCFs: (1) a VCF containing variants
+      |that were phased, and (2) a VCF containing variants that were not phased, such that all variants in the original
+      |file are in one of two output VCFs.
+      |
+      |The VCF with phased variants will have the first variant of a haplotype block with a genotype "M\F", while the
+      |subsequent variants in that block having "M|F" as their genotype.  This first variant in a block can optionally
+      |have "M|F".
+      |
+    """,
+  group=ClpGroups.Phasing
+)
+class HapCutToVcf
+( @arg(flag="v", doc="The original VCF provided to HAPCUT.") val vcf: PathToVcf,
+  @arg(flag="i", doc="The output file from HAPCUT.") val input: FilePath,
+  @arg(flag="o", doc="The output VCF with variants that were phased by HAPCUT.") val output: PathToVcf,
+  @arg(flag="u", doc="The output VCF with variants that were not phased by HAPCUT.") val unphasedOutput: PathToVcf,
+  @arg(flag="m", doc="Allow missing fields in the VCF header.") val allowMissingFieldsInVcfHeader: Boolean = true,
+  @arg(flag="r", doc="The first variant context in a block should have '/' instead of '|'") val readBackedPhasingFormat: Boolean = true
+) extends FgBioTool with LazyLogging {
+  Io.assertReadable(vcf)
+  Io.assertReadable(input)
+  Io.assertCanWriteFile(output)
+
+  override def execute(): Unit = {
+    val vcfReader      = new VCFFileReader(vcf.toFile, false)
+    val phasedWriter   = makeWriter(output, vcfReader, addPhasedHeaderLines=true)
+    val unphasedWriter = makeWriter(unphasedOutput, vcfReader, addPhasedHeaderLines=false)
+
+    val iterator = new MergingIterator(input, vcfReader, readBackedPhasingFormat)
```

Comment:
I know it's a private class but `MergingIterator` is a horrible name for this class!  How about `HapCutAndVcfMergingIterator`?